app/assets/javascripts/classes/reports/EditReportViewModel.ts

```diff
@@ -110,8 +110,8 @@ class EditReportViewModel {
             dataType: "json",
             data: JSON.stringify(request),
         })
-        .fail(() => {
-            this.alertMessage(EditReportViewModel.ERROR_MESSAGE);
+        .fail((data) => {
+            this.alertMessage(`${EditReportViewModel.ERROR_MESSAGE} Code ${data.status}. Details: ${data.responseText}`);
```

Comment:
This should use INTL type responses to format the result for the user.

Reply:
Here's the PR where Brandon did this for query processing:

https://github.com/ArpNetworking/metrics-portal/pull/133

Reply:
This is turning into a pretty extensive refactor. Fast-follow okay, to separate concerns? Other PR is already out: #292